src/CreatorTokenFactory.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {CreatorToken} from "src/CreatorToken.sol";
+import {CTBondingCurve} from "src/CTBondingCurve.sol";
+import {IERC20} from "openzeppelin/token/ERC20/IERC20.sol";
+import {IShowtimeVerifier, Attestation} from "src/lib/IShowtimeVerifier.sol";
+
+contract CreatorTokenFactory {
+  struct DeploymentConfig {
+    string name;
+    string symbol;
+    string tokenURI;
+    address creator;
+    uint256 creatorFee;
+    address admin;
+    uint256 adminFee;
+    address referrer;
+    IERC20 payToken;
+    uint128 basePrice;
+    uint128 linearPriceSlope;
+    uint128 inflectionPrice;
+    uint32 inflectionPoint;
+    bytes32 attestationDigest;
+  }
+
+  event CreatorTokenDeployed(
+    CreatorToken indexed creatorToken, CTBondingCurve indexed bondingCurve, DeploymentConfig config
+  );
+
+  error CreatorTokenFactory__DeploymentNotVerified();
+  error CreatorTokenFactory__InvalidAttestation();
+
+  bytes public constant DEPLOY_TYPE =
+    "DeploymentConfig(string name,string symbol,string tokenURI,address creator,uint256 creatorFee,address admin,uint256 adminFee,address referrer,address payToken,uint128 basePrice,uint128 linearPriceSlope,uint128 inflectionPrice,uint32 inflectionPoint,bytes32 attestationDigest)";
+
+  bytes32 public constant DEPLOY_TYPE_HASH = keccak256(DEPLOY_TYPE);
+
+  IShowtimeVerifier public immutable VERIFIER;
+
+  // TODO: dev comment explaining this is the domain separator from the verifier that technically
+  // can change if chainId or
+  // the verifier contract address change (latter can't happen if we keep the verifier hardcoded)
+  bytes32 private constant DOMAIN_SEPARATOR =
+    0x0c3be79fa914a526eb7e8ec47178f2ca6f81e90fb14c3e335600dc3ec6fa1a6f;
```

Comment:
TODO: Investigate whether this changes between base mainnet & base testnet. If so, we need to add it as an immutable var configured at deployment.